client/session.go

```diff
@@ -1334,8 +1339,8 @@ func (s *session) streamBlocksBatchFromPeer(
 			break
 		}
 
-		id := result.Elements[i].ID
-		if id != batch[i].id {
+		id := ts.BinaryID(result.Elements[i].ID)
+		if id.Hash() != batch[i].id.Hash() {
```

Comment:
Hmm don't we need a `bytes.Equal` alternative here?  Isn't `type Hash [...]byte`?  I thought you couldn't compare two `[]byte` slices?